First pass at background colors page (#1350)
grommet/hpe-design-system: aries-site/src/pages/foundation/background-colors-guidance.mdx

import { BasicLayoutExample, LayeredLayoutExample } from '../../examples';
import { Example } from '../../layouts';

> The overall design philosophy is not to use a lot of borders to distinguish content sections. This helps reduce visual clutter that the eye and brain of the user must sift through.

Reviewer: ```suggestion
The overall design philosophy is to distinguish content sections with the strategic use of color
and avoid the use of excessive borders. This helps reduce visual clutter which a reader's eye 
and brain must sift through.
```

> 

Reviewer: ```suggestion

The application of background colors creates a visual hierarchy supporting the information 
hierarchy for the page. When applying background colors to content or layout, it is helpful 
to think about the content's role and relation to the page's information hierarchy.
```

> There are two approaches to how background colors should be used: [Basic](#basic) or [Layered](#layered). See guidance for each approach below.

Reviewer: ```suggestion
There are two approaches to how background colors should be used: [Basic](#basic) or [Layered](#layered). Explore guidance for applying each approach below.
```

## Basic

> The basic approach to background color is oriented around keeping the overall user interface focused and un-cluttered. Separation of sections in the page is accomplished primarily via typography and spacing (via margins and padding).

Reviewer: ```suggestion
In this approach, background color is oriented to keep the overall user interface focused and 
clutter-free. The background color is held consistent throughout the page and content section 
separation is accomplished primarily through the use of typography and spacing (via margins 
and padding).
```

> An app-wide sidebar or header could be distinguished by another color with opposite darkness, for instance a dark blue persistent navigation sidebar. If there is a section of the page with secondary content, such as help text, background-contrast could be used.

Reviewer: ```suggestion
In the basic approach, visual distinction for a header or an app-wide sidebar from the 
page's primary background is created by applying a second color with significant contrast 
to `background`. For instance, a dark blue, [persistent navigation sidebar]() creates the desired contrast, 
achieving visual separation of the sidebar from the primary content.
```

> 

Reviewer: ```suggestion

If there is a page section presenting secondary content, such as help text, <span style={{whiteSpace: 'nowrap'}}>`background-contrast`</span>
may be used to create effective visual separation.
```

> - **background** - the default background color when no other background color is specified
> - **background-contrast** - could be used to indicate secondary content
> - **background (dark)** or another color (dark) - could be used to indicate a primary navigation sidebar or header

Reviewer: ```suggestion
- **background** - The default background color when no other background color is specified.
- **background-contrast** - Used to indicate secondary content.
- **background (dark)** or another color (dark) - Used to indicate a [primary navigation sidebar](/templates/persistent-navigation) or [header](/components/header).
```

> 

Reviewer: ```suggestion

Refer to the [background palette](/foundation/color#background-palette) for definition 
of each background color namespace and value.
```

<Example
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/foundation/background-colors/BasicLayoutExample.js"
  width="100%"
  showResponsiveControls={['laptop', 'mobile']}
  screenContainer
>
  <BasicLayoutExample />
</Example>

## Layered

> The layered approach to background color is oriented towards more complex content situations, using background colors to help separate content sections in addition to typography and margins.

Reviewer: ```suggestion
The layered approach to background color is oriented towards more complex 
content situations. In addition to typography and spacing, background colors
are used to create separate content sections and promote focus on primary content.
```

> In this approach, the overall background color should be `background-back` and specific content sections that should be the primary focus of attention should use `background-front`.

Reviewer: ```suggestion
In this approach, the overall background color should be `background-back` 
and specific content sections which should be the primary focus of attention 
should use <span style={{whiteSpace: 'nowrap'}}>`background-front`</span>.
```

> For example, a page with a header, some summary content, and then a grid of Cards would use `background-front` for each Card and then `background-back` for everything else.

Reviewer: ```suggestion
For example, a page with a [header](/components/header), summary content, and a grid of 
[cards](/components/card) would use `background-front` for each card and 
`background-back` for everything else.
```

> For a multi-column page layout, `background-front` could be used to distinguish the column that should be more in focus.

Reviewer: ```suggestion
or a multi-column page layout, `background-front` could be used to distinguish the column which should be the primary focus.
```

> From an information architecture as well as from a component layout perspective, `background-back` should not be used within a context that already has `background-front`. For instance, a primary navigation header should not use `background-front` above a main content section that uses `background-back`.

Reviewer: ```suggestion
From an information architecture, as well as from a component layout 
perspective, `background-back` should not be used within a context which 
already has <span style={{whiteSpace: 'nowrap'}}>`background-front`</span>. 
For instance, a primary navigation header should not use <span style={{whiteSpace: 'nowrap'}}>`background-front`</span> 
above a main content section that uses `background-back`.
```

> 

Reviewer: ```suggestion

From an information architecture, as well as from a component layout 
perspective, `background-back` should not be used within a context which 
already has <span style={{whiteSpace: 'nowrap'}}>`background-front`</span>. 
For instance, a primary navigation header should not use <span style={{whiteSpace: 'nowrap'}}>`background-front`</span> 
above a main content section that uses `background-back`.
```

> - **background-back** - the overall background color
> - **background-front** - the background color of content that should be the primary focus
> - **background-contrast** - could be used to indicate a secondary content area within a large content section
> - **background (dark)** or another color (dark) - could be used to indicate a primary navigation sidebar or header

Reviewer: ```suggestion
- **background-back** - The overall background color.
- **background-front** - The background color for content which should be the primary focus.
- **background-contrast** - Could be used to indicate a secondary content area within a large content section.
- **background (dark)** or another color (dark) - Could be used to indicate a primary navigation sidebar or header.
```

> 

Reviewer: ```suggestion

Refer to the [background palette](/foundation/color#background-palette) for definition 
of each background color namespace and value.
```
